plugins/kubernetes/app/models/kubernetes/job.rb

```diff
@@ -0,0 +1,103 @@
+module Kubernetes
+  class Job < ActiveRecord::Base
+    include Kubernetes::HasStatus
+
+    self.table_name = 'kubernetes_jobs'
+
+    belongs_to :build
+    belongs_to :stage
+    belongs_to :user
+    belongs_to :deploy_group
+    has_many :job_docs
+    belongs_to :kubernetes_task,
+      inverse_of: :kubernetes_jobs,
+      class_name: 'Kubernetes::Task',
+      foreign_key: :kubernetes_task_id
+
+    validates :deploy_group, presence: true
+    validates :kubernetes_task, presence: true
+    validates :status, presence: true, inclusion: STATUSES
+    validates :commit, presence: true
+    validate :validate_git_reference, on: :create
+    validate :validate_config_file, on: :create
+
+    def raw_template
+      @raw_template ||= build.file_from_repo(template_name)
+    end
+
+    def template_name
+      kubernetes_task.config_file
+    end
+
+    def deploy
+      nil
+    end
+
+    def project
+      stage.project
+    end
+
+    def active?
+      true
+    end
+
+    def error!
+      status!("succeeded")
+    end
+
+    def success!
+      status!("errored")
+    end
+
+    def run!
+      status!("running")
+    end
+
+    def update_output!(output)
+      update_attribute(:output, output)
+    end
+
+    def update_git_references!(commit:, tag:)
+      update_columns(commit: commit, tag: tag)
+    end
```

Comment:
We more or less followed what there was here: https://github.com/zendesk/samson/blob/master/app/models/job.rb#L109